src/nvim/eval.c

```diff
@@ -1833,12 +1833,15 @@ static char_u *ex_let_one(char_u *arg, typval_T *const tv,
       int opt_type;
       long numval;
       char *stringval = NULL;
+      const char *s = NULL;
 
       const char c1 = *p;
       *p = NUL;
 
       varnumber_T n = tv_get_number(tv);
-      const char *s = tv_get_string_chk(tv);  // != NULL if number or string.
+      if (tv->v_type != VAR_SPECIAL) {
```

Comment:
`VAR_BOOL` is relevant change but not ported here. Need vim-patch:8.2.0111.

Reply:
OK, I'll add it.

Reply:
@janlazo after surveyed the vim-patch:8.2.0111, I found it needs more patches (patches before 8.2.0111) to be ported to finish this PR.
Should I need to add all of them? or this PR could be done without adding those patches.

Reply:
It will be harder to port 8.2.0111 and related patches if this PR is merged now.

Reply:
I got it, I will take some time to trace back and add patches.